streams/src/test/java/org/apache/kafka/streams/state/internals/CachingKeyValueStoreTest.java

```diff
@@ -243,6 +245,50 @@ public void shouldThrowIfTryingToDoPutIfAbsentClosedCachingStore() throws Except
         store.putIfAbsent("b", "c");
     }
 
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerExceptionOnPutWithNullKey() {
+        store.put(null, "c");
```

Comment:
The `CachingKeyValueStore` doesn't throw an NPE here, rather `org.apache.kafka.common.utils.Bytes.LexicographicByteArrayComparator.compare` does. We probably should add `Objects.requireNonNull(...)` to `CachingKeyValueStore#put(..)` etc